saleor/graphql/account/utils.py

```diff
@@ -73,46 +75,74 @@ class Meta:
 
     @classmethod
     def clean_instance(cls, info, instance):
-        super().clean_instance(info, instance)
-        if not instance.is_staff:
-            raise ValidationError(
-                {
-                    "id": ValidationError(
-                        "Cannot delete a non-staff user.",
-                        code=AccountErrorCode.DELETE_NON_STAFF_USER,
-                    )
-                }
-            )
+        errors = defaultdict(list)
+        cls.check_if_users_can_be_deleted(info, [instance], "id", errors)
+        cls.check_if_requestor_can_manage_users(info, [instance], "id", errors)
+        cls.check_if_removing_left_not_manageable_permissions(
+            info, [instance], "id", errors
+        )
+        if errors:
+            raise ValidationError(errors)
 
-        cls.check_if_requestor_can_manage_user(info, instance)
-        cls.check_if_removing_left_not_manageable_permissions(instance)
+    @classmethod
+    def check_if_users_can_be_deleted(cls, info, instances, field, errors):
+        not_staff_users = set()
+        for user in instances:
+            if not user.is_staff:
+                not_staff_users.add(user)
+            try:
+                super().clean_instance(info, user)
+            except ValidationError as error:
+                errors["ids"].append(error)
+
+        if not_staff_users:
+            user_pks = [
+                graphene.Node.to_global_id("User", user.pk) for user in not_staff_users
+            ]
+            msg = "Cannot delete a non-staff users."
+            code = AccountErrorCode.DELETE_NON_STAFF_USER
+            params = {"users": user_pks}
+            errors[field].append(ValidationError(msg, code=code, params=params))
 
     @classmethod
-    def check_if_requestor_can_manage_user(cls, info, instance):
-        if get_out_of_scope_users(info.context.user, [instance]):
-            msg = "You can't manage this user."
+    def check_if_requestor_can_manage_users(cls, info, instances, field, errors):
```

Comment:
If possible I would avoid passing `info` to utility functions. In this case, only `user` is needed from `info`. To make it less confusing with `users` we could consistently call it `current_user`.

Reply:
Also maybe `current_user` would sound better than `requestor`.

Reply:
And the third thing, we're missing a docstring here.